src/main/java/org/folio/circulation/resources/RequestByInstanceIdResource.java

```diff
@@ -364,33 +358,63 @@ private void createInstanceLevelRequests(RoutingContext routingContext) {
   }
 
   private Result<InstanceRequestRelatedRecords> combineWithUnavailableItems(
-    Result<Map<Item, DateTime>> itemDueDateMapResult,
-    Result<Map<Item, Integer>> itemRequestQueueSizeMapResult,
+    Map<Item, DateTime> itemDueDateMap,
     InstanceRequestRelatedRecords records){
 
     return of(() -> {
-      if (itemDueDateMapResult.succeeded() && itemRequestQueueSizeMapResult.succeeded()) {
-        Map<Item, DateTime> itemDueDateMap = itemDueDateMapResult.value();
-        Map<Item, Integer> itemQueueSizeMap = itemRequestQueueSizeMapResult.value();
+        Map<Item, RequestQueue> itemQueueMap = records.getItemRequestQueueMap();
 
-        if (itemDueDateMap == null && itemQueueSizeMap == null) {
+        if (itemDueDateMap == null && itemQueueMap == null) {
           return records;
         }
 
-        //Sort the map
+        //transform itemQueueMap to itemQueueSizeMap
+        Map<Item, Integer> itemQueueSizeMap = itemQueueMap.entrySet()
+          .stream()
+          .collect(Collectors.toMap(
+            Map.Entry::getKey,
+            entry -> entry.getValue().size()
+            ));
+
+        //Sort the itemQueueSize map
         Map<Item, Integer> sortedMap = sortRequestQueues(itemQueueSizeMap, itemDueDateMap,
           records.getInstanceLevelRequest().getPickupServicePointId());
 
         ArrayList<Item> finalOrdedList = new ArrayList<>(sortedMap.keySet());
 
         //put the items that it wasn't able to retrieve loans or RequestQueues for on the bottom of the list
-        finalOrdedList.addAll(records.getItemsWithoutLoans());
-        finalOrdedList.addAll(records.getItemsWithoutRequests());
+        if (records.getItemsWithoutLoans() != null)
+          finalOrdedList.addAll(records.getItemsWithoutLoans());
+        if (records.getItemsWithoutRequests() != null)
+          finalOrdedList.addAll(records.getItemsWithoutRequests());
         records.setSortedUnavailableItems(finalOrdedList);
-      }
       return records;
     });
   }
+
+  private Result<Collection<Item>> validateRequester(Map<Item, RequestQueue> itemRequestQueueMap, UUID requesterId) {
+    if (!itemRequestQueueMap.isEmpty()) {
+      Collection<RequestQueue> requestQueues = itemRequestQueueMap.values();
+      for (RequestQueue queue : requestQueues) {
+        final Optional<Request> matchingRequest = queue.getRequests()
+          .stream()
+          .filter(request -> request.isOpen()
+                           && Objects.equals(request.asJson().getString(REQUESTER_ID_FIELD),
```

Comment:
good question. The Request already had a property to get the Requester that we could use, but in my tests it's always NULL. I wasn't sure if that is due to the test fixture setup or if in real life this method may or may not return NULL, so I erred on the safe side by going to the JSON directly.  I can fix the test fixture if necessary.  @marc-johnson, do you think it's a test-setup issue?

Reply:
Actually, `Request.getUserId()` returns the `requesterId`.

Reply:
As far as `getRequester()` being `null`. I could see that in a test, but the `requesterId` should never be. I didn't notice `getUserId()` already returned the `requesterId` or I'd just have suggested using that instead.

Reply:
I'll use getUserId()

Reply:
I imagine this is because not all places that use requests fetch the requesting user. The queue is probably one of those places.

The request in a queue should probably be different to a request being used specifically.